src/consumption.cpp

```diff
@@ -1861,10 +1881,14 @@ bool player::consume( item_location loc, bool force )
         return false;
     }
     item &target = *loc;
+    item_pocket *parent_pocket = nullptr;
+    if( loc.has_parent() ) {
+        parent_pocket = loc.parent_item()->contained_where( target );
+    }
     bool wielding = is_wielding( target );
     bool worn = is_worn( target );
     const bool inv_item = !( wielding || worn );
-    if( consume( target, force ) ) {
+    if( consume( target, force, parent_pocket ) ) {
```

Comment:
Is there a reason for passing along the pocket and doing the checks inside instead of just unsealing the pocket if consume returns true?

Reply:
Consume returns true only if all items are consumed. A pocket is unsealed as long as any item is consumed.